python/pyarrow/tests/test_serialization.py

```diff
@@ -52,6 +52,9 @@
     sparse = None
 
 
+pytestmark = pytest.mark.filterwarnings("ignore:'pyarrow:DeprecationWarning")
```

Comment:
Hmm... is this another pytest magic? What does it do exactly? Filter these warnings only for this test module?

Reply:
Indeed, this is a way to suppress (ignore) all DeprecationWarnings that start with "pyarrow" in this test module. That's easier than catching all individual warnings, since this full file is about serialization and thus deprecated. 
(and therefore I added a separate file to explicitly test the warnings are raised)

Will add a short comment about it